plugins/outputs/warp10/warp10.go

```diff
@@ -0,0 +1,293 @@
+	req.Header.Set("X-Warp10-Token", w.Token)
+	req.Header.Set("Content-Type", "text/plain")
+
+	resp, err := w.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
+		return fmt.Errorf(w.WarpURL + ": " + w.HandleError(string(body)))
+	}
+
+	return nil
+}
+
+func buildTags(tags []*telegraf.Tag) []string {
+
+	tagsString := make([]string, len(tags)+1)
+	indexSource := 0
+	for index, tag := range tags {
+		tagsString[index] = fmt.Sprintf("%s=%s", tag.Key, tag.Value)
+		indexSource = index
+	}
+	indexSource++
+	tagsString[indexSource] = fmt.Sprintf("source=telegraf")
+	sort.Strings(tagsString)
+	return tagsString
+}
+
+func buildValue(v interface{}) (string, error) {
+	var retv string
+	switch p := v.(type) {
+	case int64:
+		retv = intToString(int64(p))
+	case string:
+		retv = fmt.Sprintf("'%s'", strings.Replace(p, "'", "\\'", -1))
+	case bool:
+		retv = boolToString(bool(p))
+	case uint64:
+		retv = uIntToString(uint64(p))
+	case float64:
+		retv = floatToString(float64(p))
+	default:
+		retv = "'" + strings.Replace(fmt.Sprintf("%s", p), "'", "\\'", -1) + "'"
+	}
+	return retv, nil
+}
+
+func intToString(inputNum int64) string {
+	return strconv.FormatInt(inputNum, 10)
+}
+
+func boolToString(inputBool bool) string {
+	return strconv.FormatBool(inputBool)
+}
+
+func uIntToString(inputNum uint64) string {
+	return strconv.FormatUint(inputNum, 10)
+}
+
+func floatToString(inputNum float64) string {
+	return strconv.FormatFloat(inputNum, 'f', 6, 64)
+}
+
+// SampleConfig get config
+func (w *Warp10) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description get description
+func (w *Warp10) Description() string {
+	return "Configuration for Warp server to send metrics to"
+}
+
+// Close close
+func (w *Warp10) Close() error {
+	return nil
+}
+
+func init() {
+	outputs.Add("warp10", func() telegraf.Output {
+		return &Warp10{}
+	})
+}
+
+// HandleError read http error body and return a corresponding error
+func (w *Warp10) HandleError(body string) string {
+	if body == "" {
+		return "Empty return"
+	}
+
+	if strings.Contains(body, "Invalid token") {
+		return fmt.Sprintf("Invalid token: %v", w.Token)
+	}
+
+	if strings.Contains(body, "Write token missing") {
+		return "Write token missing"
+	}
+
+	if strings.Contains(body, "Token Expired") {
+		return fmt.Sprintf("Token Expired: %v", w.Token)
+	}
+
+	if strings.Contains(body, "Token revoked") {
+		return fmt.Sprintf("Token revoked: %v", w.Token)
+	}
+
+	if strings.Contains(body, "exceed your Monthly Active Data Streams limit") || strings.Contains(body, "exceed the Monthly Active Data Streams limit") {
+		reg := regexp.MustCompile(`Monthly Active Data Streams limit(?: for application.*)? \((\d+)(.\d+)?(E-\d)?\)`)
+		parts := reg.FindStringSubmatch(body)
+		limit := "-1"
+		if len(parts) > 1 {
+			limit = parts[1]
+		}
+		return fmt.Sprintf("MADS exceeded: %v", limit)
```

Comment:
This will look strange if the regex doesn't match, it may not be worth the complexity to retrieve this information.  If you want to keep the regex though we should compile the regex at package level.  If we switch to using the status line for the error message though I think we can just return the full error.